RELEASE_NOTES.md

```diff
@@ -7,6 +7,7 @@
 ## Upgrading
 
 - Some minimal dependencies have been bumped, so you might need to adjust your dependencies accordingly.
+- Improve documentation of the frequenz.channels.experimental.Pipe
```

Comment:
This is not really an upgrade step, so I would put it into `New Features` (even if it is not a new feature either :D, sometimes I create a new section `# Enhancements` in these cases, but I think it can also be left out of the release notes, as I don't think anyone will go check out the `Pipe` docs just because it is in the changelog).